prow/config/jobs.go

```diff
@@ -231,6 +231,9 @@ type Periodic struct {
 	Cron string `json:"cron,omitempty"`
 	// Tags for config entries
 	Tags []string `json:"tags,omitempty"`
+	// IntervalWaitsForCompletion waits for current job to finish before for interval calculation starts
+	// This allows to add variance to job scheduling
+	IntervalWaitsForCompletion bool `json:"interval_waits_for_completion,omitempty"`
```

Comment:
so to use this new feature, the job will be:
```
interval: 2h
interval_waits_for_completion: true
```
which are two flags, and the second flag works only with `interval`. Would it be easier to make this new flag also a time duration just like `interval`?

Reply:
Right. The current proposal introduces a new option, `interval_waits_for_completion`, that changes the behavior of the `interval` type periodic.

We may instead want a new type of periodics, such as `interval_with_drift` (please suggest names) that takes a time duration if that's the desired API.

The new `interval_waits_for_completion` option seemed more natural to me than a new type of specialized `interval` periodics, but that's probably only personal preference. I also find that a new option describes also better what it does.

I don't really have a strong preference, and I'd be happy to go with whatever you folks chose.

Reply:
I think it's easier to use and easier to understand with a new separate option. How about name it `interval_between_runs`? this feels straightforward to me, wdyt?

Reply:
The `interval_between_run` name would be a bit confusing since with the `interval` type periodic jobs, you already have exactly the specified interval between consequent runs. Chosing a name is difficult :-)
How do you like `minimum_interval`, to highlight the specified interval is guaranteed?

Reply:
Agree that interval should already behave as interval between runs, and unfortunately it's hard to change something like this. And yeah naming something is hard, let's just go with your proposal and not blocked by naming. Please add more comments on these two fields to make sure future reader understand the difference

Reply:
Ok, I'll make the change as soon as I'm in front of a computer. I'll go with `minimum_interval` for the new option name.

Reply:
Done. Sadly the code is now more involved due to the new option.